### PR TITLE
feat(ST-101): add update exam api endpoint

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
## Description of Change

- Added PATCH `classrooms\:classroomId\exams\:examId` API route
- Added service functions

## Associated Ticket Link(s)

- https://trello.com/c/oT0WHjYr

## Related Pull Request(s)

- N/A

### CODE REVIEW

File: src/exams/exams.service.ts [hunk 1]
```diff
@@ -31,4 +31,14 @@ export class ExamsService {
...
   }
+
+  async updateExam(classroomId: number, examId: number, updateExamDto: UpdateExamDto) {
+    const classroom = await this.classroomsRepository.findOneOrFail({ id: classroomId });
+    const exam = await this.examsRepository.findOneOrFail({ id: examId, classroom: classroom.id });
+
+    this.examsRepository.assign(exam, updateExamDto);
+    await this.em.persistAndFlush(exam);
+
+    return exam;
```
Comment (ishmam-iqbal-sazim) [inline src/exams/exams.service.ts:42]: Let's put this inside a repository method named updateOne and call that here

File: src/exams/exams.dtos.ts [hunk 2]
```diff
@@ -12,3 +12,8 @@ export class CreateExamDto {
   @Transform(({ value }) => new Date(value))
   date!: Date;
 }
+
+export class UpdateExamDto extends CreateExamDto {
+  @IsNumber()
+  id!: number;
```
Comment (ishmam-iqbal-sazim) [inline src/exams/exams.dtos.ts:18]: Do you need the id?

File: src/exams/exams.dtos.ts [hunk 3]
```diff
@@ -12,3 +12,8 @@ export class CreateExamDto {
   @Transform(({ value }) => new Date(value))
   date!: Date;
 }
+
+export class UpdateExamDto extends CreateExamDto {
```
Comment (ishmam-iqbal-sazim) [inline src/exams/exams.dtos.ts:16]: The attributes in update dto should be optional

File: src/exams/exams.dtos.ts [hunk 4]
```diff
@@ -12,3 +12,18 @@ export class CreateExamDto {
...
+  id?: number;
+
+  @IsString()
+  title?: string;
+
+  @IsString()
+  instruction?: string;
+
+  @IsDate()
+  @Transform(({ value }) => new Date(value))
```
Comment (ishmam-iqbal-sazim) [inline src/exams/exams.dtos.ts:27]:
```suggestion
  @Type(() => Date)
```

Does this not work?

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comment

**Review by ragib-hossain-sazim — APPROVED** [review]

**Pull request merged**